adr/0010-scripts-actions.md

```diff
@@ -0,0 +1,43 @@
+# 10. Scripts -> Actions
+
+Date: 2023-01-18
+
+## Status
+
+Accepted
+
+## Context
+
+Originally, the `scripts` noun was added to components to allow us to move hard-coded init business logic out of the codebase and into the package system. At the time there was only a `before` and `after` section with simply an array entry per command. Later, `prepare` was added as a way to do some thing during `zarf package create`. As teams began to find new ways to use the capabilities, their limitations became more obvious.
```

Comment:
I think something about what's being done with scripts is important.  It feels like we're saying we're replacing scripts, but no comment about how to update those nouns within a zarf.yaml or what it looks like for dealing with zarf.yaml versions from earlier versions of Zarf.

Reply:
that is actually covered in the consequences section, where I think it is most approrpiate